src/v/storage/segment_appender.h

```diff
@@ -32,7 +32,20 @@ namespace storage {
 /// Appends data to a log segment. It can be subclassed so
 /// other classes can add behavior and still be treated as
 /// an appender.
-/// Note: The functions in this call cannot be called concurrently.
+///
+/// The append() functions in this class take different input types to
+/// append but all return future<> and have the same general semantics:
+/// After the future<> for an append() call returns, the data has been
+/// logically appended to the segment in memory, but may not be, or not
+/// fully be flushed to disk and in general hasn't been fsynced. After
+/// the future from append() results, a subject flush() returns a future
+/// whose resolution indicates that all prior appends have been flushed
+/// and fsync'd on disk.
+///
+/// NOTE: Only one append() may be progress at one time. I.e., it is not
+/// safe to call append() before the prior append() call has resolved.
+/// However, there are no requirements around concurrent flushing: flush
+/// may be called even if other flushes or appends are in progress.
 class segment_appender {
```

Comment:
The deleted line was from Alex a long time ago but since then @dotnwat has loosened the requirement so that it the main restriction is that there cannot be "concurrent" `append` requests but flush/flush and append/flush concurrency is OK.

This new comment is my understanding of the current restriction.